deps/prepare_submodule.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+set -x
+COMP="$1"
+GIT_TAG=""
+DAOS_ROOT="$(realpath "$(realpath "$(dirname "${BASH_SOURCE[0]}")")/..")"
+
+get_tag()
+{
+  pushd "${DAOS_ROOT}" || exit 1
+  git submodule update --init --recursive
+  submodule="${DAOS_ROOT}/deps/${COMP}"
+  # shellcheck disable=SC1090,SC2283
+  source <(grep = <(grep -A10 '\[commit_versions\]' "${DAOS_ROOT}/utils/build.config"))
+  GIT_TAG="${!COMP}"
+
+  if [ -z "${GIT_TAG}" ]; then
+    echo "Empty tag for ${COMP}"
+    exit 1
+  fi
+
+  eval "${COMP}"=""
+  # shellcheck disable=SC1090,SC2283
+  source <(grep = <(grep -A10 '\[repos\]' "${DAOS_ROOT}/utils/build.config"))
+  REPO="${!COMP}"
```

Comment:
FYI I think this should work too to avoid polluting the current env with all the names (might need to fix quoting here to make shellcheck happy)
```suggestion
  REPO="$(source <(grep = <(grep -A10 '\[repos\]' "${DAOS_ROOT}/utils/build.config")); echo "${!COMP}")"
```